### PR TITLE
🚀 feat: Integrate Amazon Bedrock Support

### PR DESCRIPTION
**Pull request by Sunwood-ai-labs**
## Overview
This PR adds support for Amazon Bedrock models to our LLM integration, enhancing our AI capabilities with Claude 3 models (Opus, Sonnet, Haiku).

## Key Changes
- Implement AWS credentials retrieval for Bedrock
- Add Bedrock model initialization and handling
- Include Claude 3 models options for Bedrock
- Adjust token limits specifically for Bedrock models
- Update chat action to support dynamic model selection
- Add necessary dependencies for Amazon Bedrock integration

## Detailed Changes
1. `app/lib/.server/llm/api-key.ts`: 
   - Add `getAWSCredentials` function to fetch AWS access keys and region

2. `app/lib/.server/llm/constants.ts`: 
   - Define `MAX_TOKENS_BEDROCK` constant (4096) for Bedrock models

3. `app/lib/.server/llm/model.ts`: 
   - Implement `getBedrockModel` function for Bedrock model initialization
   - Update `getModel` function to handle Bedrock provider

4. `app/lib/.server/llm/stream-text.ts`: 
   - Use Bedrock-specific token limit (`MAX_TOKENS_BEDROCK`)

5. `app/routes/api.chat.ts`: 
   - Update chat action to support model selection in the request body
   - Set proper Content-Type header for the response

6. `app/utils/constants.ts`: 
   - Add Bedrock model options (Claude 3 Opus, Sonnet, Haiku)

7. `package.json`: 
   - Add `@ai-sdk/amazon-bedrock` dependency (version 0.0.30)

8. `pnpm-lock.yaml`: 
   - Update with new dependencies including AWS SDK packages

### CODE REVIEW

**Comment by Sunwood-ai-labs** [comment]
I implemented Bedrock based on #531 of https://github.com/coleam00/bolt.new-any-llm!

File: README.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 [![Bolt.new: AI-Powered Full-Stack Web Development in the Browser](./public/social_preview_index.jpg)](https://bolt.new)
 
+# Bolt.new Fork by Cole Medin
+
+This fork of bolt.new allows you to choose the LLM that you use for each prompt! Currently you can use OpenAI, Anthropic, Ollama, or Groq models - and it is easily extended to use any other model supported by the Vercel AI SDK! See instructions below for running this locally and extending to include more models.
+
```
Comment (rrfaria) [inline README.md:6]: - you added on readme and I don't know if creators will accept this:

File: app/lib/.server/llm/model.ts [hunk 2]
```diff
@@ -1,9 +1,77 @@
...
+  const apiKey = getAPIKey(env, provider);
+  
+  switch (provider) {
+    case 'Anthropic':
+      return getAnthropicModel(apiKey, model);
+    case 'OpenAI':
+      return getOpenAIModel(apiKey, model);
+    case 'Groq':
+      return getGroqModel(apiKey, model);
+    case 'OpenRouter':
```
Comment (rrfaria) [inline app/lib/.server/llm/model.ts:72]: - [add gemini](https://github.com/stackblitz/bolt.new/pull/531/files#diff-bf2719e734530c6c4ed0f8c45e7e6c572f16e374a700fc4e10c97ea417ef2699R5)

**Review by rrfaria — APPROVED** [review]
Perhaps you need change the branch name to pass on CI, something like feat/multi-llm because of semantic Pull request

**Comment by Sunwood-ai-labs** [comment]
I changed the branch name, but the content is the same as
- #733 !

**Pull request closed without merging**